charts/capi-runtime-extensions/defaultclusterclasses/nutanix-cluster-class.yaml

```diff
@@ -0,0 +1,708 @@
+            path: /etc/kubernetes/manifests/kube-vip.yaml
+        initConfiguration:
+          nodeRegistration:
+            kubeletExtraArgs:
+              cloud-provider: external
+              eviction-hard: nodefs.available<10%,nodefs.inodesFree<5%,imagefs.available<15%,memory.available<100Mi,imagefs.inodesFree<10%
+              tls-cipher-suites: ${TLS_CIPHER_SUITES=TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256,TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256}
+        joinConfiguration:
+          nodeRegistration:
+            kubeletExtraArgs:
+              cloud-provider: external
+              eviction-hard: nodefs.available<10%,nodefs.inodesFree<5%,imagefs.available<15%,memory.available<100Mi,imagefs.inodesFree<10%
+              tls-cipher-suites: ${TLS_CIPHER_SUITES=TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256,TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256}
+        postKubeadmCommands:
+          - echo export KUBECONFIG=/etc/kubernetes/admin.conf >> /root/.bashrc
+          - echo "after kubeadm call" > /var/log/postkubeadm.log
+        preKubeadmCommands:
+          - echo "before kubeadm call" > /var/log/prekubeadm.log
+          - hostnamectl set-hostname "{{ ds.meta_data.hostname }}"
+        useExperimentalRetryJoin: true
+        verbosity: 10
+---
+apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+kind: NutanixClusterTemplate
+metadata:
+  labels:
+    cluster.x-k8s.io/provider: infrastructure-nutanix
+  name: ${CLUSTER_CLASS_NAME}-nct
+  namespace: ${NAMESPACE}
+spec:
+  template:
+    spec:
+      failureDomains: []
+---
+apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+kind: NutanixMachineTemplate
+metadata:
+  labels:
+    cluster.x-k8s.io/provider: infrastructure-nutanix
+  name: ${CLUSTER_CLASS_NAME}-cp-nmt
+  namespace: ${NAMESPACE}
+spec:
+  template:
+    spec:
+      bootType: ${NUTANIX_MACHINE_BOOT_TYPE=legacy}
+      cluster:
+        name: ${NUTANIX_PRISM_ELEMENT_CLUSTER_NAME}
+        type: name
+      image:
+        name: ${NUTANIX_MACHINE_TEMPLATE_IMAGE_NAME}
+        type: name
+      memorySize: ${NUTANIX_MACHINE_MEMORY_SIZE=4Gi}
+      providerID: nutanix://${CLUSTER_NAME}-m1
+      subnet:
+        - name: ${NUTANIX_SUBNET_NAME}
+          type: name
+      systemDiskSize: ${NUTANIX_SYSTEMDISK_SIZE=40Gi}
+      vcpuSockets: ${NUTANIX_MACHINE_VCPU_SOCKET=2}
+      vcpusPerSocket: ${NUTANIX_MACHINE_VCPU_PER_SOCKET=1}
+---
+apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+kind: NutanixMachineTemplate
+metadata:
+  labels:
+    cluster.x-k8s.io/provider: infrastructure-nutanix
+  name: ${CLUSTER_CLASS_NAME}-md-nmt
+  namespace: ${NAMESPACE}
+spec:
+  template:
+    spec:
```

Comment:
We should ideally also add:
- project
- additionalCategories
- opus

Reply:
yes we need to figure out how to add required vs optional fields so that user can choose either way. this can be added in subsequent PRs